docs/design-tech/sync.md

```diff
@@ -1,89 +1,118 @@
 
-![handle_server_change](handle_server_change.svg)
+The metadata for deleted files must be eventually pruned on clients so that the amount of space taken on users' devices scales with the number of not-deleted files rather than the total number of files ever created. Deleted files cannot be immediately pruned because the client needs to remember to push the deletion to the server on the next sync. The server keeps track of all file metadata forever (though it deletes contents of deleted files) so that clients can sync after being offline indefinitely and still receive deletion events without the complexity of the server needing to know which clients exist and which updates they have each received. We are not concerned about storage usage on the server growing with the number of total files ever created.
 
-[src](https://docs.google.com/drawings/d/18mBM2NLZJ3Ykz64-i9BaINmnGpVfp5wXvaiwqphH7zM/edit)
+## Design Concerns
+Sync v3 operates in the context of many concerns. We'll discuss the design first in terms of how the concerns are addressed one concern at a time, then in terms of the client and server routines that address all these concerns all together. These are the concerns we'll discuss:
+* file deletion
+* invariant violation resolution
+* consistency
 
-### `core -> server`
+### File Deletion
+As mentioned, files have various states of deletion:
+* not deleted (the file has `false` value for `deleted` and no deleted ancestors)
+* explicitly deleted (the file has `true` value for `deleted`)
+* implicitly deleted (the file has `false` value for `deleted` but has an explicitly deleted ancestor)
+* pruned (there is no mention of file on disk, although it once existed)
 
-Pushing local changes from core is much more trivial. Core will ask `LocalChangesRepo` what changed and perform the
-corresponding network action. If a file was renamed, for example, it will send a `/rename` request to the server.
+First we'll justify the distinction between deleted and pruned. If the file deletion system is working properly, then after a user deletes a file on a device and syncs that device, that device will prune the file and any other devices that sync will prune that file. What's required to make that happen is that the client which deletes the file remembers that the file exists (and is deleted) until that client syncs its changes. Once the changes are synced, the client can prune the file knowing that the server will remember the deletion forever and will inform the other clients of the deletion. Note that deleted documents can have their contents pruned immediately - it is only important to remember the metadata until the deletion is pushed.
```

Comment:
todo: we can delete file contents for implicitly deleted files, but we might have to re-pull them during a sync